core/strings/builder.odin

```diff
@@ -146,6 +146,12 @@ to_string :: proc(b: Builder) -> string {
 	return string(b.buf[:])
 }
 
+// cast the builder byte buffer to a cstring and return it. assert that the buffer ends with a terminating nul byte.
+to_cstring :: proc(b: Builder) -> cstring {
+	assert(b.buf[len(b.buf)-1] == 0)
+	return cstring(&b.buf[0])
+}
```

Comment:
This is a very bad assumption to make. Don't do this.

Reply:
I asserted to let users know this is a requirement for the conversion to work and to avoid creating a string of undefined length. Wouldn't it be even more dangerous to not assert here? How would this be done the odin way?

Reply:
To do this safely, it would have been better to check if the last value was `0`, and if not, append `0`. And if the length of the buffer is zero or could not append, return `nil`.